internal/servers/plugin/v3/plugin.go

```diff
@@ -130,8 +130,13 @@ func (s *Server) Sync(req *pb.Sync_Request, stream pb.Plugin_SyncServer) error {
 			syncErr = fmt.Errorf("failed to sync records: %w", err)
 		}
 	}()
-
+	var err error
 	for msg := range msgs {
+		msg, err = s.Plugin.OnBeforeSend(ctx, msg)
+		if err != nil {
+			syncErr = fmt.Errorf("failed before sending message: %w", err)
```

Comment:
I can't recall now but why we `break` here and not `return` like in all other places below

Reply:
The reason I did it this way was that I didn't want it to be treated as an internal error like in the other `return` cases, but I think it's all the same really

Reply:
I think I'll change it to return just to be clear